src/devices/stepper.rs

```diff
@@ -0,0 +1,526 @@
+//
+//
+// Licensed under the MIT license:
+//   <LICENSE or http://opensource.org/licenses/MIT>
+// This file may not be copied, modified, or distributed except according
+// to those terms.
+//
+
+use crate::{AttachCallback, DetachCallback, GenericPhidget, Phidget, Result, ReturnCode};
+use phidget_sys::{self as ffi, PhidgetHandle, PhidgetStepperHandle as StepperHandle};
+use std::{mem, os::raw::c_void, ptr};
+
+/// The function type for the safe Rust position change callback.
+pub type PositionChangeCallback = dyn Fn(&Stepper, f64) + Send + 'static;
+/// The function type for the safe Rust velocity change callback.
+pub type VelocityChangeCallback = dyn Fn(&Stepper, f64) + Send + 'static;
+/// The function type for the safe Rust stop callback.
+pub type StoppedCallback = dyn Fn(&Stepper) + Send + 'static;
+
+/// Phidget Stepper sensor
+pub struct Stepper {
+    // Handle to the sensor for the phidget22 library
+    chan: StepperHandle,
+    // Double-boxed StepperCallback, if registered
+    cb: Option<*mut c_void>,
+    // Double-boxed attach callback, if registered
+    attach_cb: Option<*mut c_void>,
+    // Double-boxed detach callback, if registered
+    detach_cb: Option<*mut c_void>,
+}
+
+/// ControlMode for stepper
+#[derive(Copy, Clone)]
+pub enum ControlMode {
+    /// Step: Control the motor by setting a target position.
+    Step,
+    /// Run: Control the motor by selecting a target velocity (sign indicates direction). The motor will rotate continuously in the chosen direction.
+    Run,
+}
+
+impl Stepper {
+    /// Create a new Stepper sensor.
+    pub fn new() -> Self {
+        let mut chan: StepperHandle = ptr::null_mut();
+        unsafe {
+            ffi::PhidgetStepper_create(&mut chan);
+        }
+        Self::from(chan)
+    }
+
+    /// Get a reference to the underlying sensor handle
+    pub fn as_channel(&self) -> &StepperHandle {
+        &self.chan
+    }
+
+    /// Set enable failsafe
+    pub fn set_enable_failsafe(&self, failsafe_time: u32) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_enableFailsafe(self.chan, failsafe_time)
+        })?;
+        Ok(())
+    }
+
+    /// Add position offset
+    pub fn get_add_position_offset(&self, position_offset: f64) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_addPositionOffset(self.chan, position_offset)
+        })?;
+        Ok(())
+    }
+
+    /// Set reset failsafe
+    pub fn set_reset_failsafe(&self) -> Result<()> {
+        ReturnCode::result(unsafe { ffi::PhidgetStepper_resetFailsafe(self.chan) })?;
+        Ok(())
+    }
+    /// Set acceleration
+    pub fn set_acceleration(&self, acceleration: f64) -> Result<()> {
+        ReturnCode::result(unsafe {
+            ffi::PhidgetStepper_setAcceleration(self.chan, acceleration)
+        })?;
+        Ok(())
+    }
+
+    /// Get acceleration
+    pub fn get_acceleration(&self) -> Result<f64> {
```

Comment:
Rust getters don't use `get_`. Just make this `acceleration(&self)`

Same for all getters.